primeqa/qg/processors/data_loader.py

```diff
@@ -0,0 +1,47 @@
+from primeqa.qg.processors.passage_qg.tydiqa_processor import TydiQADataset
+from datasets import Dataset
+from primeqa.qg.processors.passage_qg.squad_processor import SquadDataset
+from primeqa.qg.processors.table_qg.wikisql_processor import WikiSqlDataset
+
+class QGDataLoader():
+	def __init__(self, 
+				tokenizer,
+				dataset_name='wikisql', 
+				input_max_len=512,
+				target_max_len=32
+				):
+		self.tokenizer = tokenizer
+		self.dataset_name = dataset_name
+		self.input_max_len = input_max_len
+		self.target_max_len = target_max_len
+		
+	def convert_to_features(self, example_batch):
+		input_encodings = self.tokenizer.batch_encode_plus(example_batch['input'], 
+										pad_to_max_length=True, max_length=self.input_max_len)
+		target_encodings = self.tokenizer.batch_encode_plus(example_batch['question'], 
+										pad_to_max_length=True, max_length=self.target_max_len)
+		encodings = {
+			'input_ids': input_encodings['input_ids'], 
+			'attention_mask': input_encodings['attention_mask'],
+			'target_ids': target_encodings['input_ids'],
+			'target_attention_mask': target_encodings['attention_mask']
+		}
+
+		return encodings
+
+	def create(self, data_split='train'):
+		if self.dataset_name == 'wikisql':
+			data = WikiSqlDataset()
+		elif self.dataset_name in ['squad', 'squad_v2']:
+			data = SquadDataset(self.dataset_name)	
+		elif self.dataset_name in ['tydiqa']:	
+			data = TydiQADataset(self.dataset_name)
+		else:
+			raise NotImplementedError("this data not supported")
```

Comment:
If these arguments are given as choices then this line can't be reached. Missing "is"